domain/schema/model/sql/0018-machine.sql

```diff
@@ -0,0 +1,184 @@
+CREATE TABLE machine (
+    uuid              TEXT PRIMARY KEY,
+    machine_id        TEXT NOT NULL,
+    net_node_uuid     TEXT NOT NULL,
+    life_id           INT NOT NULL,
+    base              TEXT,
+    nonce             TEXT,
+    container_type_id INT,
+    password_hash     TEXT,
+    clean             BOOLEAN,
+    force_destroyed   BOOLEAN,
+    placement         TEXT,
+    agent_started_at  DATETIME,
+    hostname          TEXT,
+    CONSTRAINT        fk_machine_net_node
+        FOREIGN KEY   (net_node_uuid)
+        REFERENCES    net_node(uuid),
+    CONSTRAINT        fk_machine_life
+        FOREIGN KEY   (life_id)
+        REFERENCES    life(id),
+    CONSTRAINT        fk_machine_container_type
+        FOREIGN KEY   (container_type_id)
+        REFERENCES    container_type(id)
+);
+
+CREATE UNIQUE INDEX idx_machine_id
+ON machine (machine_id);
+
+CREATE UNIQUE INDEX idx_machine_net_node
+ON machine (net_node_uuid);
+
+CREATE TABLE container_type (
+    id    INT PRIMARY KEY,
+    value TEXT NOT NULL
+);
+
+INSERT INTO container_type VALUES
+    (0, 'lxd');
+
+CREATE TABLE machine_constraint (
+    machine_uuid     TEXT NOT NULL,
+    constraint_uuid  TEXT NOT NULL,
+    CONSTRAINT       fk_machine_constraint_machine
+        FOREIGN KEY  (machine_uuid)
+        REFERENCES   machine(uuid),
+    CONSTRAINT       fk_machine_constraint_constraint
+        FOREIGN KEY  (constraint_uuid)
+        REFERENCES   "constraint"(uuid),
+    PRIMARY KEY (machine_uuid, constraint_uuid)
+);
+
+CREATE TABLE machine_principal (
```

Comment:
We can drop this. We will know the units on the machine via the link between `net_node` and `unit`.